bin/generate-php-sync-issue.mjs

```diff
@@ -0,0 +1,501 @@
+/**
+ * External dependencies
+ */
+
+import Octokit from '@octokit/rest';
+import fs from 'fs';
+
+import { fileURLToPath } from 'url';
+import nodePath, { dirname } from 'path';
+
+function getArg( argName ) {
+	const arg = process.argv.find( ( _arg ) =>
+		_arg.startsWith( `--${ argName }=` )
+	);
+	return arg ? arg.split( '=' )[ 1 ] : null;
+}
+
+const OWNER = 'wordpress';
+const REPO = 'gutenberg';
+const MAX_MONTHS_TO_QUERY = 4;
+
+const IGNORED_PATHS = [
+	'lib/load.php', // plugin specific code.
+	'lib/experiments-page.php', // experiments are plugin specific.
+	'packages/e2e-tests/plugins', // PHP files related to e2e tests only.
+	'packages/block-library', // this is handled automatically.
+];
+
+const MAX_NESTING_LEVEL = 3;
+
+const DEBUG = !! getArg( 'debug' );
+
+const __filename = fileURLToPath( import.meta.url );
+const __dirname = dirname( __filename );
+
+const authToken = getArg( 'token' );
+const stableWPRelease = getArg( 'wpstable' );
+
+async function main() {
+	if ( ! authToken ) {
+		console.error(
+			'Error. The --token argument is required. See: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token'
+		);
+		process.exit( 1 );
+	}
+
+	if ( ! stableWPRelease ) {
+		console.error(
+			'Error. The --wpstable argument is required. It should be the current stable WordPress release (e.g. 6.4).'
+		);
+		process.exit( 1 );
+	}
+
+	const sinceArg = getArg( 'since' );
+	let since;
+
+	if ( sinceArg ) {
+		if ( validateDate( sinceArg ) ) {
+			since = sinceArg;
+		} else {
+			console.error(
+				`Error: The --since argument cannot be more than ${ MAX_MONTHS_TO_QUERY } months from the current date.`
+			);
+			process.exit( 1 );
+		}
+	} else {
+		console.error(
+			`Error. The --since argument is required (e.g. YYYY-MM-DD). This should be the date of the previous release's final RC.`
```

Comment:
I think I got this wrong when we discussed this earlier 🤦‍♀️ It should be all changes from the final version of Gutenberg that's bundled with the last WordPress release. So for 6.5, it would be any changes from GB 16.8 onwards, as 16.7 was the last version of the plugin included in 6.4. There's a table with the included version numbers [here](https://developer.wordpress.org/block-editor/contributors/versions-in-wordpress/).

There will be some overlap from bugs fixes, etc but this should capture all changes. Sorry for the confusion!

Reply:
✅ Updated.